keras/utils/data_utils.py

```diff
@@ -641,35 +664,45 @@ def data_generator_task():
                 except Exception as e:
                     # Can't pick tracebacks.
                     # As a compromise, print the traceback and pickle None instead.
-                    if self._use_multiprocessing:
-                        traceback.print_exc()
-                        setattr(e, '__traceback__', None)
-                    elif not hasattr(e, '__traceback__'):
-                        setattr(e, '__traceback__', sys.exc_info()[2])
+                    traceback.print_exc()
```

Comment:
Why remove the traceback when using threads?

Reply:
This is a question better asked to @de-vri-es who made commit [4a58b178073f0ba3b166220f7ebd7d56149bfb20](https://github.com/fchollet/keras/commit/4a58b178073f0ba3b166220f7ebd7d56149bfb20)

Reply:
the exception is put in the queue and rethrown in the main thread. Printing the trackback twice is no good. Also, if someone de decides to rethrow the exception you don't want the trackbacks printer at all.

The only reason I kept the print in multiprocessing=True is that the trackback cant be pickled, and cant be put in an inter-process queue.

Unconditionally printing the trackback is not a good idea in my opinion.

Reply:
I meant "if someone de decides to catch the exception you don't want the trackback printed at all" . I should 't do this from my phone apparantly...

Reply:
It looks like you undid my work in a merge conflict here.

Reply:
I don't believe so. The test for multiprocessing=True has been taken out of the while loop for clarity (there is now a code path for multiprocessing and one for multithreading). The diff displayed by Github appears misleading however. @de-vrie-es, would you mind looking at the files side by side and tell me if you agree? I would greatly appreciate it.

Reply:
Oh sorry, I didn't see the full diff (I really shouldn't have continued on my phone). I see the `if/else` is still there but much earlier. Looks good =]